samcli/commands/deploy/guided_context.py

```diff
@@ -88,7 +89,7 @@ def guided_parameter_overrides(self):
     # pylint: disable=too-many-statements
     def guided_prompts(self, parameter_override_keys):
         default_stack_name = self.stack_name or "sam-app"
-        default_region = self.region or "us-east-1"
+        default_region = self.region or get_session().get_config_variable("region") or "us-east-1"
```

Comment:
this seems redundant. we are already defaulting region to us-east-1. what is the benefit of getting region from boto session?

Reply:
That's literally the only change this pull request makes. It looks for the user's default region in the aws config file (found in ~/.aws/config) or in the environment variables (as AWS_DEFAULT_REGION). If neither is found It defaults to us-east-1.

Reply:
I understand what it does. Just want to understand why we need this? Are you trying to prevent some corner case failure here? or is it to just improve general UX?

Reply:
Improve user experience